src/components/tetris/tetris.tsx

```diff
@@ -1,127 +1,79 @@
+  boardChange: ReplaySubject<TetrisMatrix>
+  getColumnCount(): number
+  getRowCount(): number
+}
 
 interface TetrisProps {
+  game: Game
   staticPath: string
-  onGameOver: OnGameOver
 }
 
 interface TetrisState {
-  totalScore: number
-  gainedScore?: number
-  currentLevel: number
+  score: number
 }
 
 type Ctx = CanvasRenderingContext2D
 type Canvas = HTMLCanvasElement
 
-const ONE_SECOND = 1000
-
 export class Tetris extends Component<TetrisProps, TetrisState> {
-  private game: Game
-  private columnCount = 10
-  private rowCount = 16
-  private onBoardChange: OnBoardChange
-  private onScoreChange: OnScoreChange
-  private onLevelChange: OnLevelChange
   private board: TetrisMatrix
   private blockSVG: SVGElement
   private readonly canvasRef = React.createRef<HTMLCanvasElement>()
-
-  state: TetrisState = {
-    totalScore: 0,
-    gainedScore: undefined,
-    currentLevel: 1,
-  }
-
-  componentWillUnmount() {
-    this.onBoardChange = () => undefined
-    this.onScoreChange = () => undefined
-  }
+  private subscriptions: Subscription[] = []
 
   async componentDidMount() {
     this.resizeCanvasToParentSize()
     window.onresize = () => this.resizeCanvasToParentSize()
 
-    this.onBoardChange = (board: TetrisMatrix) => { this.board = board }
-    this.onScoreChange = async (gainedScore: number, totalScore: number) => {
-      this.setState({ gainedScore, totalScore })
-      await wait(ONE_SECOND)
-      this.setState({ gainedScore: undefined })
-    }
-    this.onLevelChange = (currentLevel: number) =>
-      this.setState({ currentLevel })
-
-    this.game = new Game(
-      this.onBoardChange,
-      this.onScoreChange,
-      this.onLevelChange,
-      this.columnCount,
-      this.rowCount,
-    )
-
-    const xhr = new XMLHttpRequest()
-    const url = this.props.staticPath + "/block.svg"
-    xhr.open("GET", url, true)
-    xhr.overrideMimeType("image/svg+xml")
-    xhr.onload = () => {
-      if (xhr.responseXML && xhr.responseXML.documentElement) {
-        this.blockSVG = xhr.responseXML.documentElement as any as SVGElement
-      }
-    }
-    xhr.send("")
+    await this.loadShapeBlock()
 
     const canvasRef = this.canvasRef
     const ctx = this.ctx
     const renderFrame = () => {
-      if (canvasRef.current && ctx ) {
+      if (canvasRef.current && ctx) {
         this.renderGame(canvasRef.current, ctx)
       }
       window.requestAnimationFrame(renderFrame)
     }
-
     window.requestAnimationFrame(renderFrame)
 
-    await this.game.start()
-    /* tslint:disable */
-    console.log(this.board)
-    /* tslint:enable */
-    this.props.onGameOver(this.state.totalScore)
-  }
-
-  rotate() {
-    this.game.rotate()
-  }
-
-  left() {
-    this.game.left()
+    const { game } = this.props
+    this.subscriptions.push(
+      game.boardChange.subscribe(board => (this.board = board)),
+    )
   }
 
-  right() {
-    this.game.right()
+  componentWillUnmount() {
+    this.subscriptions.map(s => s.unsubscribe())
```

Comment:
@szokodiakos Since you are not doing anything with the result this should be a `.forEach` instead of a `.map`